Packs/QRadar/Integrations/QRadar_v2/QRadar_v2.py

```diff
@@ -0,0 +1,1961 @@
+        return offense
+
+
+def try_poll_offense_events_with_retry(
+    client, offense_id, query_status, search_id, max_retries=None
+):
+    """
+    Polls search until the search is done (completed/canceled/error), and then returns the search result
+    will retry up to max_retries consecutive failures
+    """
+    if not max_retries:
+        max_retries = EVENTS_FAILURE_LIMIT
+    i = 1
+    failures = 0
+    while not (query_status in TERMINATING_SEARCH_STATUSES or failures >= max_retries):
+        try:
+            if is_reset_triggered(client.lock):
+                return []
+
+            raw_search = client.get_search(search_id)
+            query_status = raw_search.get("status")
+            # failures are relevant only when consecutive
+            failures = 0
+            if query_status in TERMINATING_SEARCH_STATUSES:
+                raw_search_results = client.get_search_results(search_id)
+                print_debug_msg(
+                    f"Events fetched for offense {offense_id}.", client.lock
+                )
+                events = raw_search_results.get("events", [])
+                for event in events:
+                    try:
+                        for time_field in EVENT_TIME_FIELDS:
+                            if time_field in event:
+                                event[time_field] = epoch_to_iso(event[time_field])
+                    except TypeError:
+                        continue
+                return events
+            else:
+                # prepare next run
+                i = i % 60 + EVENTS_INTERVAL_SECS
+                if i >= 60:  # print status debug every fetch sleep (or after)
+                    print_debug_msg(
+                        f"Still fetching offense {offense_id} events, search_id: {search_id}.",
+                        client.lock,
+                    )
+                time.sleep(EVENTS_INTERVAL_SECS)
+        except Exception:
+            failures += 1
+    return []
+
+
+def try_create_search_with_retry(client, events_query, offense, max_retries=None):
+    if not max_retries:
+        max_retries = EVENTS_FAILURE_LIMIT
+    failures = 0
+    search_created_successfully = False
+    query_status = ""
+    search_id = ""
+    while not search_created_successfully and failures <= max_retries:
+        try:
+            raw_search = client.search(events_query)
+            search_res_events = deepcopy(raw_search)
+            search_res_events = filter_dict_non_intersection_key_to_value(
+                replace_keys(search_res_events, SEARCH_ID_NAMES_MAP),
+                SEARCH_ID_NAMES_MAP,
+            )
+            search_id = search_res_events.get("ID")
+            query_status = search_res_events.get("Status")
+            search_created_successfully = True
+        except Exception:
+            failures += 1
+    if failures >= EVENTS_FAILURE_LIMIT:
+        raise DemistoException(f"Unable to create search for offense: {offense['id']}.")
```

Comment:
the exception msg is not included, please add it so we will have more information about what happened.